source/ConversationPanel.cpp

```diff
@@ -339,6 +340,33 @@ void ConversationPanel::Goto(int index, int choice)
 			// Update any altered government reputations.
 			player.CheckReputationConditions();
 		}
+		else if(conversation.IsPayment(node))
+		{
+			int64_t payment = conversation.Payment(node);
+			int64_t account = player.Accounts().Credits();
+			// If the payment is negative and the player doesn't have enough
+			// in their account, then the player's credits are reduced to 0.
```

Comment:
If the player has a large negative account, and the payment is small positive (or even negative), then the player gains money in this action.

Reply:
If the player has a large negative account, and the payment is small positive (or even negative), then the player gains money in this action. The player's account is never allowed to go negative at any point in the game. From the comment on Account::AddCredits:
> // Give the player credits (or pass  negative number to subtract). **If subtracting,
> // the calling function needs to check that this will not result in negative credits.**

Reply:
Ok, but if we change the else-statement to `else if(account > 0)` then we are also safe if some bug somewhere else in the code does causes the account to go negative. (Or if we do allow negative numbers in the future).